JS/Airport.js

```diff
@@ -1,120 +1,83 @@
+    
+     getPassengerPlanes() {
+        var passengerPlaneList = [];
+        for (let plane of this.planes) {
+            if (plane instanceof PassengerPlane) {passengerPlaneList.push(plane);}
         }
-        return x;
+        return passengerPlaneList;
     }
 
     getMilitaryPlanes() {
-        let militaryPlanes = [];
+        let militaryPlanesList = [];
         this.planes.forEach(plane => {
-            if (plane instanceof MilitaryPlane) {//if
-                militaryPlanes.push(plane);
+            if (plane instanceof MilitaryPlane) {
+                militaryPlanesList.push(plane);
             }
-            //else
-            else{}
         });
-//return
-        return militaryPlanes;
+        return militaryPlanesList;
     }
 
+
     getPassengerPlaneWithMaxPassengersCapacity() {
-        let passengerPlanes = this.getPasPl();
-        let planeWithMaxCapacity = passengerPlanes[0];
-        for (let i = 0; i < passengerPlanes.length; i++) {
-            if (passengerPlanes[i].getPassengersCapacity() >     planeWithMaxCapacity.getPassengersCapacity()) {
-                planeWithMaxCapacity = passengerPlanes[i];
-            }
-        }
-        return planeWithMaxCapacity;
+        let passengerPlanes = this.getPassengerPlanes();
+        return passengerPlanes.reduce(function(a, b) {
+         return (a.getPassengersCapacity() > b.getPassengersCapacity() ? a : b);
+        });
     }
 
-
-
-
-
-
-
-    getTransportMilitaryPlanes(){
-        let transportMilitaryPlanes = [];
-        let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-        if (militaryPlanes[i].getMilitaryType() == MilitaryType.TYPE_TRANSPORT) {
-        transportMilitaryPlanes.push(militaryPlanes[i]);
-        }
-        }
-        return transportMilitaryPlanes;
+    getTransportMilitaryPlanes() {
+       let militaryPlanesArray = this.getMilitaryPlanes();
+       let transportMilitaryPlanes = militaryPlanesArray.filter(el => el.getMilitaryType() == MilitaryType.TRANSPORT);
+       return transportMilitaryPlanes;
     }
 
-
     getBomberMilitaryPlanes()
     {
-        let bomberMilitaryPlanes = [];
         let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-            if (militaryPlanes[i].getMilitaryType()=== MilitaryType.BOMBER) {
-                bomberMilitaryPlanes.push(militaryPlanes[i]);
-            }
-        }
+        let bomberMilitaryPlanes = militaryPlanes.filter (el => el.getMilitaryType() == MilitaryType.BOMBER);
         return bomberMilitaryPlanes;
     }
 
     constructor(planes) {
         this.planes = planes;
     }
 
-
     getExperimentalPlanes() {
-        let experimentalPlanes  = [];
-        this.planes.forEach(plane => {
-            if (plane instanceof experimentalPlane) {//if
+        let experimentalPlanes = [];
+        this.planes.forEach((plane) => {
```

Comment:
Try to rewrite this code using `filter()` array method.

Reply:
Done